File: src/test/resources/person.pl
% handle a java Person object

:- dynamic id/1.

name_of(Id, Name) :- jpl_call(@(Id), 'getName', [], Name).
age_of(Id, Age) :- jpl_call(@(Id), 'getAge', [], Age).
gender_of(Id, Gender) :- jpl_call(@(Id), 'getGender', [], Gender).
spouse_of(Id, Spouse) :- jpl_call(@(Id), 'getSpouse', [], Spouse).

person(Id, Name, Age, Gender, Spouse) :-
	id(Id),
	name_of(Id, Name),
	age_of(Id, Age),
	gender_of(Id, Gender),
	spouse_of(Id, Spouse).



% one possible way to handle event notification  
% myassert(A) :- jpl_call(EventListener, 'execute', ['predicate/arity'], @(null)), assert(A).

File: src/test/java/au/edu/rmit/csit/swijadex/Person.java
package au.edu.rmit.csit.swijadex;

public class Person {

	private String name;
	private int age;
	private String gender;
	private Person spouse;
	
	public Person(String name, int age, String gender) {
		this.name = name;
		this.age = age;
		this.gender = gender;
	}

	public Person(String name, int age) {
		this(name, age, null);
	}
	
	public String getName() {
		return this.name;
	}
	
	public int getAge() {
		return this.age;
	}
	
	public String getGender() {
		return this.gender;
	}
	
	public Person getSpouse() {
		return this.spouse;
	}

	public void setName(String name) {
		this.name = name;
	}

	public void setAge(int age) {
		this.age = age;
	}
	
	public void setGender(String gender) {
		this.gender = gender;
	}

	public void setSpouse(Person spouse) {
		this.spouse = spouse;
	}
	
	public String toString() {
		return "{"
				+ "name: " + this.getName()
				+ ", age: " + this.getAge()
				+ ", gender: " + this.getGender()
				+ ", spouse: " + this.getSpouse()
			+ "}";
	}

	public void dumpDetails() {
		System.out.println(this);
	}

}
